dipy/workflows/reconst.py

```diff
@@ -21,6 +21,330 @@
+
+            if not save_metrics:
+                save_metrics = ['rtop', 'laplacian_signal', 'msd', 'qiv', 'rtap', 'rtpp']
+
+            # print('data.shape (%d, %d, %d, %d)' % data.shape)
+
+            radial_order = 6
+            
+            # How do I save the rtop in a useful form? A picture? A numpy array?
+            # Save them as nifti image
+
+            if model_type is 'laplacian':
+                map_model_aniso = mapmri.MapmriModel(gtab, radial_order=radial_order,
+                                                     laplacian_regularization=True,
+                                                     laplacian_weighting=.2)
+                mapfit_aniso = map_model_aniso.fit(data)
+
+            elif model_type is 'positivity':
+
+                map_model_aniso = mapmri.MapmriModel(gtab, radial_order=radial_order,
+                                                     laplacian_regularization=False,
+                                                     positivity_constraint=True)
+                mapfit_aniso = map_model_aniso.fit(data)
+
+            elif model_type is 'both':
+                map_model_aniso = mapmri.MapmriModel(gtab, radial_order=radial_order,
+                                                     laplacian_regularization=True,
+                                                     laplacian_weighting=.05,
+                                                     positivity_constraint=True)
+                mapfit_aniso = map_model_aniso.fit(data)
+
+            # Not sure where to get affine or metadata?
+            if 'rtop' in save_metrics:
+                r = mapfit_aniso.rtop()
+                rtop = nib.nifti1.Nifti1Image(r.astype(np.float32), affine)
+                nib.save(rtop, out_rtop)
+
+            if 'laplacian_signal' in save_metrics:
+                l = mapfit_aniso.norm_of_laplacian_signal()
+                lap = nib.nifti1.Nifti1Image(l.astype(np.float32), affine)
+                nib.save(lap, out_lapnorm)
+
+            if 'msd' in save_metrics:
+                m = mapfit_aniso.msd()
+                msd = nib.nifti1.Nifti1Image(m.astype(np.float32), affine)
+                nib.save(msd, out_msd)
+
+            if 'qiv' in save_metrics:
+                q = mapfit_aniso.qiv()
+                qiv = nib.nifti1.Nifti1Image(q.astype(np.float32), affine)
+                nib.save(qiv, out_qiv)
+
+            if 'rtap' in save_metrics:
+                r = mapfit_aniso.rtap()
+                rtap = nib.nifti1.Nifti1Image(r.astype(np.float32), affine)
+                nib.save(rtap, out_rtap)
+
+            if 'rtpp' in save_metrics:
+                r = mapfit_aniso.rtpp()
+                rtpp = nib.nifti1.Nifti1Image(r.astype(np.float32), affine)
+                nib.save(rtpp, out_rtpp)
+
+            logging.info('MAPMRI saved in {0}'.
+                         format(os.path.dirname(out_rtpp)))
+            
+
+class ReconstMAPMRILaplacian(ReconstMAPMRIFlow):
```

Comment:
@Garyfallidis : is this the right way to do things? You create an interface for every setting of options to the model? I thought this was supposed to create a CLI that allows you to input key-word arguments?